src/build.js

```diff
@@ -43,6 +45,74 @@ export async function generateBuildMap(configObj) {
   return await iterateDir(configObj.inputDir);
 }
 
+export async function generateBundleOutputObj(buildMap) {
+  const filenames = buildMap.map(buildPair => buildPair[0].pathname);
+
+  function PRESMPlugin() {
+    return {
+      name: 'PRESMPlugin',
+      resolveId(source) {
+        if (filenames.includes(source)) {
+          return source;
+        }
+        return null;
+      },
+      // If source is in buildmap, return source code
+      load(id) {
+        const sourceArr = buildMap.filter(
+          buildPair => buildPair[0].pathname === id
+        );
+        if (sourceArr) {
+          return sourceArr[0][1];
+        }
+        return null;
+      },
+    };
+  }
+
+  const inputOptions = {
+    input: filenames,
+    plugins: [PRESMPlugin()],
```

Comment:
This construction kind of gives me a O_o ... can you explain why a function is used here vs just inlining it or using a plain object above?

Reply:
If it's a function: Maybe `createPlugin` instead. `PascalCase` is usually reserved for types and classes, and functions use `camelCase`.

Reply:
Gotcha, yup was following the [example here](https://rollupjs.org/guide/en/#a-simple-example).  I do agree that having it be a function is strange, updated it to be an object with more conventional naming as well